src/main/java/emissary/core/InterruptibleCharSequence.java

```diff
@@ -0,0 +1,33 @@
+package emissary.core;
+
+public class InterruptibleCharSequence implements CharSequence {
+    CharSequence inner;
+
+    public InterruptibleCharSequence(CharSequence inner) {
+        super();
+        this.inner = inner;
+    }
+
+    @Override
+    public char charAt(int index) {
+        if (Thread.currentThread().isInterrupted()) {
+            throw new InterruptedByTimeoutException("Interrupted reading CharSequence");
```

Comment:
I'm not sure this new exception type is needed: we can observe that the current thread is interrupted, but can't definitely assert that it's due to a timeout.  I think a standard `InterruptedException` might be sufficient.

Reply:
I believe that any exception thrown from within `charAt` needs to be an instance of a `RuntimeException`. `InterruptedException` only extends `Exception`, which would require that `charAt` declare an exception can be thrown.